nncf/ptq/data/dataloader.py

```diff
@@ -0,0 +1,157 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+from typing import Callable
+from typing import TypeVar
+from abc import ABC
+from abc import abstractmethod
+
+
+DataSource = TypeVar('DataSource')
+
+
+class NNCFDataLoader(ABC):
+    """
+    The `NNCFDataLoader` provides input samples for the model.
```

Comment:
Will it always provide only inputs?

Reply:
Not applicable. The code was refactored.